src/main/java/wolox/training/models/Book.java

```diff
@@ -71,6 +72,8 @@ public String getAuthor() {
     }
 
     public void setAuthor(String author) {
+        Preconditions.checkArgument(!author.isEmpty() && author != null,
```

Comment:
Change all checks to first check != null and then the isEmpty, as this, as almost any language, evaluates from left to right, so you'll end up with a NullPointerException like this.